src/morphology/english/getAdjectiveForms.js

```diff
@@ -40,81 +40,98 @@ const checkWordTooLong = function( word ) {
+
+	return endsWithEst && ! estExceptions.includes( word );
 };
 
 /**
- * Checks if the input word ends with "ly".
+ * Checks if the input word can be an adverb with "-ly".
  *
- * @param {string} word The word to check.
+ * @param {string}      word            The word to check.
+ * @param {string[]}    lyExceptions    The list of words that end with -ly but are not adverbs.
  *
- * @returns {boolean} True if the word ends with "ly".
+ * @returns {boolean} True if the word can be an adverb with "-ly".
  */
-const endsWithLy = function( word ) {
+const canBeLyAdverb = function( word, lyExceptions ) {
 	const wordLength = word.length;
+
 	// Consider only words of five letters or more to be adjectives (otherwise, words like "lily" are being treated as adjectives).
-	if ( wordLength >= 5 ) {
-		return word.substring( word.length - 2, word.length ) === "ly";
+	if ( wordLength < 5 ) {
+		return false;
 	}
-	return false;
+
+	const endsWithLy = word.substring( word.length - 2, word.length ) === "ly";
+	return endsWithLy && ! lyExceptions.includes( word );
 };
 
 
 /**
  * Forms the base form from an input word.
  *
- * @param {string} word The word to build the base form for.
- * @param {Array} comparativeToBaseRegex The Array of regex-based rules to bring comparatives to base.
- * @param {Array} superlativeToBaseRegex The Array of regex-based rules to bring superlatives to base.
- * @param {Array} adverbToBaseRegex The Array of regex-based rules to bring adverbs to base.
- * @param {string[]} lyExceptions The array of words that end with "ly" but are not adverbs.
+ * @param {string}  word                            The word to build the base form for.
+ * @param {Array}   comparativeToBaseRegex          The Array of regex-based rules to bring comparatives to base.
+ * @param {Array}   superlativeToBaseRegex          The Array of regex-based rules to bring superlatives to base.
+ * @param {Array}   adverbToBaseRegex               The Array of regex-based rules to bring adverbs to base.
+ * @param {Object}  stopAdjectives                  The lists of words that are not adverbs.
+ * @param {string[]}  stopAdjectives.erExceptions   The list of words that end with -er and are not comparatives.
+ * @param {string[]}  stopAdjectives.estExceptions  The list of words that end with -est and are not comparatives.
+ * @param {string[]}  stopAdjectives.lyExceptions   The list of words that end with -ly and are not adverbs.
  *
  * @returns {string} The base form of the input word.
  */
-const getBase = function( word, comparativeToBaseRegex, superlativeToBaseRegex, adverbToBaseRegex, lyExceptions ) {
-	if ( endsWithEr( word ) ) {
+const getBase = function( word, comparativeToBaseRegex, superlativeToBaseRegex, adverbToBaseRegex, stopAdjectives ) {
+	if ( canBeComparative( word, stopAdjectives.erExceptions ) ) {
 		return {
-			base: buildOneFormFromRegex( word, comparativeToBaseRegex ),
+			base: buildOneFormFromRegex( word, comparativeToBaseRegex ) || word,
```

Comment:
I need `buildOneFormFromRegex` and its brother `buildTwoFormsFromRegex` to fail if no regex fit the input string. That is how the logic of forms building worked so far. If it would return the same word, things would get trickier, because some rules rightly return the unchanged word. 
There is room for improvement in how morphological rules are implemented now, but I guess it falls out of the scope of this PR. Do you agree?